### ISSUE TITLE
set orientation to landscape

### ISSUE THREAD

**Issue by milahu** [attah/ppm2pwg#27]
currently i dont see a way to set orientation to landscape

when i print a landscape document, its rotated by 90 degrees and cropped

```
pdf2printable \
  -f urf \
  -p 1-2 \
  --paper-size iso_a4_210x297mm \
  --resolution 600 \
  --duplex \
  --quality high \
  --antialias \
  src.pdf dst.urf

ippclient print ipp://192.168.178.197/ dst.urf
```

i tried `--media-pos top` and `--media-pos left` but no change

i tried `--paper-size iso_a4_297x210mm` but that makes it worse
the angle is correct, but the output is too small, and still cropped

### expected

with `lp` this just works
so maybe `pdf2printable` and/or `ippclient` should auto-detect the orientation

```
lp src.pdf \
  -o sides=two-sided-short-edge \
  -o PageSize=A4 \
  -o PrintQuality=600dpi \
  -d Brother_HL-L5100DN_4
```

generally:
i expect `pdf2printable` and/or `ippclient` to accept free-form key-value pairs
similar to `lp -o key=val`

then, to set orientation to landscape
i could use `-o orientation-requested=landscape`
which translates to an IPP file like

```
orientation-requested=landscape
```

or `-o 0.media-source-properties.media-source-feed-orientation=4`
which translates to an IPP file like

```ipp
{
  media-source-properties=
  {
    media-source-feed-orientation=4 # landscape
  }
}
```

related: [IPP Finishing 2.1: Orientation and Feed Orientation Extensions](https://lists.pwg.org/pub/pwg/ipp/slides/Finishings-2.1-Orientation-and-Feed-Orientation-Extensions-2016-05-17.pdf)

**Comment by attah** [comment]
Landscape printing is one of those things that, the more you read, the less sense it makes.
Non-square page data has an orientation. Printers (or their trays) have a feed direction.
Neither of those should indeed need any user setting any options.
Forcing a tiny landscape on a portrait paper is *layouting* and should not be part of the print pipeline.

What is your use-case?

- Do you have long-edge feed and your printer is imposing the image letterboxed?
- Are you lacking access to telling your printer to perform long-edge feed?
- Or are you wanting layouting?
- ...

Not counting the landscape paper size, which is of course wrong; pdf2printable should "rotate and scale to fit".
I have it on good authority that, perhaps except roll printers, i should never (have to) produce landscape raster.
So if you managed to get pdf2printable to crop something you found a bug, and i want to know what you did.

Have you tried either of those ipp snippets, and did they work?

Freeform IPP may or may not happen. Not particularly keen on making a parser, and current JSON abilities are not exactly command line friendly (and thus not exposed for input).
...and i'm also not particularly keen on getting bug reports on that the convert pipeline should have respected said freeform attributes that it by definition knows nothing about.

**Comment by milahu** [comment]
> What is your use-case?

sample: [2 pages of my pdf](https://github.com/user-attachments/files/16640537/wersindmeinefreunde.2024-03-24.booklet.pages1-2.pdf)

this was rendered from html ([bindery](https://github.com/evnbr/bindery)) with css style

```css
div.page { width: 297mm; height: 210mm; }
@page { margin: 0; }
```

`lp src.pdf` just works...

> Have you tried either of those ipp snippets

no, i have no idea how that works.
how do i specify my pdf/pwg/urf file in the ipp file?
(sorry for the stupid question...)

i also tried the [libcups](https://github.com/OpenPrinting/libcups) tools `ipptransform` and `ipptool`, but no luck

```
$ ipptransform -o page-ranges=1-2 -m image/urf \
  -o media=A4 -r 600dpi -t sgray_8 src.pdf -d ipp://192.168.178.197/
ipptransform: Unable to send print data: client-error-bad-request
```

same error with `ipptransform -m image/pwg-raster`

next i will try to reproduce the exact commands called by cups
since the cups driver seems to use a proprietary raster format

but then, `ippclient info` says my printers accepts `urf` and `pwg` formats

```
$ ippclient info ipp://192.168.178.197/
...

Make and model:
enBrother HL-L5100DN series
...

IPP versions:
1.0, 1.1, 2.0

IPP features:
airprint-1.5
wfds-print-1.0
...

Document formats (native):
application/octet-stream
image/urf
image/pwg-raster
...
```

**Comment by attah** [comment]
That does indeed break my rotate-to-fit.
I don't think you need any landscape settings at all - either the pdf is borked or my code has a bug.

**Comment by attah** [comment]
Fixed by 84b195a.
I lost the reference on some parameters that were technically not needed to be modified when changing data type.
This made rotation happen with incorrect center.
Now the original solution is restored and it produces the expected output - no landscape settings required.
